### PR TITLE
warn on unknown languages

### PR DESCRIPTION
**Pull request by Trilarion**

### CODE REVIEW

**Comment by Trilarion** [comment]
Seems to work. My inbuilt errors (Achtung, die Kurve!'s contains Javascript as language, which is not recognized as known language by us. Please check for spelling errors. and Abuse's contains Lispo as language, which is not recognized as known language by us. Please check for spelling errors.) have been reported.

If this PR passes the review, it can be undone.

File: dangerfile.js [hunk 1]
```diff
@@ -49,18 +52,32 @@ const checkRepoAdded = game => {
...
+	let languages = game.lang
+	if (!Array.isArray(languages)) languages = [languages]
+	languages = languages.filter(l => !knownLanguages.includes(l))
+	if (languages.length) {
+		warn(`${game.name}'s contains ${languages} as language, which is not recognized as known language by us. ` +
+		`Please check for spelling errors.`)
+		unknownLanguageDetected = true
+	}
+}
+
```
Comment (cxong) [inline dangerfile.js]: Please keep indenting consistent

File: dangerfile.js [hunk 2]
```diff
@@ -49,18 +52,32 @@ const checkRepoAdded = game => {
   )
 }
 
+const checkLanguageKnown = game => {
+	if (!game.lang) return
+	let languages = game.lang
+	if (!Array.isArray(languages)) languages = [languages]
+	languages = languages.filter(l => !knownLanguages.includes(l))
```
Comment (cxong) [inline dangerfile.js]:
```suggestion
	const unknownLanguages = languages.filter(l => !knownLanguages.includes(l))
```

File: dangerfile.js [hunk 3]
```diff
@@ -49,18 +52,32 @@ const checkRepoAdded = game => {
...
+	if (languages.length) {
+		warn(`${game.name}'s contains ${languages} as language, which is not recognized as known language by us. ` +
```
Comment (cxong) [inline dangerfile.js]:
```suggestion
		warn(`${game.name}'s contains "${unknownLanguages}" as language, which is not known by us. ` +
```

File: dangerfile.js [hunk 4]
```diff
@@ -12,6 +12,9 @@ let namesRemoved = []
 
 const isGame = game => /^games\/\w+\.yaml$/.test(game)
 
+let unknownLanguageDetected = false
+const knownLanguages = ['ActionScript', 'Ada', 'AngelScript', 'Assembly', 'Blitz BASIC', 'C', 'C#', 'C++', 'CoffeeScript', 'D', 'Delphi', 'Elm', 'F#', 'GDScript', 'GML', 'Go', 'Haskell', 'Haxe', 'Java', 'JavaScript', 'Kotlin', 'Lisp', 'Lua', 'Object Pascal', 'Objective-C', 'ooc', 'Pascal', 'Perl', 'PHP', 'Python', 'QBasic', 'QuakeC', 'QuickBASIC', 'Ruby', 'Rust', 'Scala', 'Swift', 'TorqueScript', 'TypeScript', 'Visual FoxPro']
```
Comment (cxong) [inline dangerfile.js]: please expand this so that every entry is in one line

File: games/t.yaml [hunk 5]
```diff
@@ -604,7 +601,7 @@
   updated: 2017-10-23
   url: https://github.com/xesf/twin-e
 
-- name: Typescript Xenon
+- name: Xenon
```
Comment (cxong) [inline games/t.yaml]:
```suggestion
- name: TypeScript Xenon
```
The name comes from the URL; please do not change

File: dangerfile.js (hunk 1 above)
Comment (Trilarion) [reply dangerfile.js]: indentation is consistent now

File: dangerfile.js (hunk 4 above)
Comment (Trilarion) [reply dangerfile.js]: is expanded

**Comment by Trilarion** [comment]
Requested changes are done. From my side it looks good now.

**Pull request merged**